src/tag-input/src/TagInput.js

```diff
@@ -166,30 +176,118 @@ const TagInput = memo(
     return (
       <Box
         aria-disabled={disabled || undefined}
-        aria-activedescendant={isFocused ? id : undefined}
+        aria-activedescendant={isFocused ? inputId : undefined}
         aria-invalid={isInvalid}
         className={cx(themedContainerClassName, className)}
         ref={ref}
         onBlur={handleBlur}
         {...boxProps}
         {...rest}
+        paddingRight={hasAutocomplete ? majorScale(3) : undefined}
       >
         {values.map(maybeRenderTag)}
-        <TextInput
-          appearance="none"
-          id={id}
-          disabled={disabled}
-          flexGrow="1"
-          height={height - 4}
-          width="auto"
-          type="text"
-          value={inputValue}
-          {...inputProps}
-          ref={inputRef}
-          onChange={handleInputChange}
-          onFocus={handleInputFocus}
-          onKeyDown={handleKeyDown}
-        />
+        <Box flexGrow="1" display="inline-block">
+          <Autocomplete
+            onChange={changedItem => {
+              addTags(changedItem)
+              setInputValue('')
+            }}
+            items={hasAutocomplete ? autocompleteItems : []}
+            id={inputId}
+            selectedItem=""
+            inputValue={inputValue}
+          >
+            {autocompleteProps => {
+              const {
+                closeMenu,
+                getInputProps,
+                getRef: autocompleteGetRef,
+                getToggleButtonProps,
+                highlightedIndex
+              } = autocompleteProps
+
+              const {
+                onBlur: autocompleteOnBlur,
+                onChange: autocompleteOnChange,
+                onKeyDown: autocompleteKeyDown,
+                ...autocompleteRestProps
+              } = getInputProps()
+
+              return (
+                <>
+                  <TextInput
+                    appearance="none"
+                    disabled={disabled}
+                    height={height - 4}
+                    width="auto"
+                    type="text"
+                    {...inputProps}
+                    {...autocompleteRestProps}
+                    value={inputValue}
+                    id={inputId}
+                    ref={textInputRef => {
+                      autocompleteGetRef(textInputRef)
+                      if (inputRef instanceof Function) {
+                        inputRef(textInputRef)
+                      } else if (inputRef) {
+                        inputRef.current = textInputRef
+                      }
+                    }}
+                    onBlur={e => {
+                      autocompleteOnBlur(e)
+                      if (inputProps.onBlur) {
```

Comment:
Is this truthy check necessary for `safeInvoke`? Not sure what the implementation in there does, but the name implies it would be safe to pass in `undefined`/`null` or other non-function values without blowing up.

Reply:
Good point! I think I can skip the check.